src/addrman.cpp

```diff
@@ -385,7 +385,12 @@ void CAddrMan::Unserialize(Stream& s_)
         LogPrint(BCLog::ADDRMAN, "addrman lost %i new and %i tried addresses due to collisions or invalid addresses\n", nLostUnk, nLost);
     }
 
-    Check();
+    const int check_code{ForceCheckAddrman()};
+    if (check_code != 0) {
+        throw std::ios_base::failure(strprintf(
+            "Corrupt data. Consistency check failed with code %s",
+            check_code));
```

Comment:
It would be nice if this mentions "addrman" or "peers.dat". Without it the error message looks too generic, it is unclear where the problem is and the user may freak out that the "corrupt data" applies to the wallet:

> terminate called after throwing an instance of 'std::ios_base::failure[abi:cxx11]'
  what():  Corrupt data. Consistency check failed with code -1: iostream error

Reply:
Can you explain what code path didn't catch the exception? Raw exceptions shouldn't be displayed to the user at all. See also https://github.com/bitcoin/bitcoin/pull/22734#issue-714811417 on steps to test this pull request.

Reply:
Uff, sorry, I remember this was discussed before. I was misled somehow by the fuzzer output (quoted above).

Reply:
No worries. Resolving for now.